### PR TITLE
get rid of new double in CIBMethod.cpp

### PR DESCRIPTION
**Pull request by fuhuifang**

### CODE REVIEW

File: src/IB/CIBMethod.cpp [hunk 1]
```diff
@@ -1402,8 +1402,9 @@ CIBMethod::constructMobilityMatrix(const std::string& /*mat_name*/,
     }
 
     // Get the position data
-    double* XW = nullptr;
-    if (rank == managing_rank) XW = new double[size];
+    std::vector<double> XW;
+    if (rank == managing_rank) XW.resize(size);
+    XW.data();
```
Comment (drwells) [inline src/IB/CIBMethod.cpp]: This statement won't have any effect alone; we should just have a blank line.

File: src/IB/CIBMethod.cpp [hunk 2]
```diff
@@ -1462,7 +1463,6 @@ CIBMethod::constructMobilityMatrix(const std::string& /*mat_name*/,
                 }
             }
         }
-        delete[] XW;
```
Comment (drwells) [inline src/IB/CIBMethod.cpp]: :)

Comment (fuhuifang) [reply src/IB/CIBMethod.cpp]: Just pushed the changes.

**Comment by boyceg** [comment]
This looks good to me. @fuhuifang can you squash into one or two commits (see, e.g., [here](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit)), and then I will merge.

**Comment by boyceg** [comment]
@amneetb, can you take a look before we merge this?

**Comment by amneetb** [comment]
@fuhuifang: If the CIB examples are building and running (serial and parallel both), then it should be fine. The changes look OK to me.

**Comment by boyceg** [comment]
This is not something that would have been introduced by this PR; it sounds like something that was already broken. @amneetb, can you take a look? @drwells, are we running parallel cases in the tests at this point?

**Comment by drwells** [comment]
We do not have any parallel test cases yet (i.e., nothing is invoked with `mpirun`/`mpiexec`).

This is most likely fallout from #380; I'll take a look.

**Comment by boyceg** [comment]
May be worth running IB/explicit/ex0 in parallel to see if it is CIB or anything IB-like.

**Comment by boyceg** [comment]
I am going to go ahead and merge because I think these changes are all fine.

**Comment by drwells** [comment]
@fuhuifang BTW: if you write 'closes #358' in a comment GitHub will automatically close the issue once this is merged.

**Comment by boyceg** [comment]
I just checked, and this PR does not fully close #358 -- there are still some uses of `new double` lurking; just noting in `CIBMethod.cpp`.

Fuhui, do you want to make a new PR that tries to finish up the rest? Do
```
$ grep "new double" `find . -name *.cpp`
```
You will see that all of this comes from stuff under the auspices of @amneetb. ;-)

**Comment by boyceg** [comment]
BTW, getting rid of `new T` is more difficult, because we initialize `SAMRAI::tbox::Pointer<T>` using `new`.

Because SAMRAI 3 now uses `std::shared_ptr<T>`, we can go back and remove any remaining uses of `new` after that update.

**Comment by drwells** [comment]
Ah, I thought the warning was printed to the screen. I see it in the log file too :)

**Comment by boyceg** [comment]
I am about to push a fix. It looks like what happened is that the member `LDataManager::d_needs_synch` was being initialized incorrectly _before_ we moved initialization to the header.

**Comment by drwells** [comment]
Yes. This issue predates #380.

**Pull request merged**